src/components/repeat/repeat.test.ts

```diff
@@ -26,4 +26,12 @@ describe('RepeatingFieldComponent - handleExpressionFieldIdUpdate', () => {
       "myValue > today() || myValue <= '1/1/1890' || myValue > useFieldValue('visit_date') || myValue < useFieldValue('visit_date')",
     );
   });
+
+  it('Should update field ID in expressions when adding repeated fields', () => {
+    const expression = "infantStatus !== 'someValue'";
+    const fieldIds = ['birthDate', 'infantStatus', 'deathDate'];
+    const updatedExpression = updateFieldIdInExpression(expression, 2, fieldIds);
+
+    expect(updatedExpression).toEqual("infantStatus_2 !== 'someValue'");
+  });
```

Comment:
@WodPachua I like the test but the scope of work begs for testing the [repeat component](https://github.com/openmrs/openmrs-esm-form-engine-lib/blob/main/src/components/repeat/repeat.component.tsx) itself and not just the helpers. You want to render a form with a "repeatable" field, and then test out things like:

1. Asserting that clicking the "Add" button clones the field at origin 
2. Test out submission (Assert that both the origin and its instances' values are submitted successfully)
3. Edit mode (Assert that a repeat field with instances is properly initialised (hydrated) as expected)